src/main/java/com/labsynch/labseer/chemclasses/rdkit/CmpdRegSDFReaderRDKitImpl.java

```diff
@@ -1,31 +1,44 @@
 package com.labsynch.labseer.chemclasses.rdkit;
 
+import java.io.File;
+import java.io.FileNotFoundException;
 import java.io.IOException;
+import java.util.Scanner;
 
 import com.labsynch.labseer.chemclasses.CmpdRegMolecule;
 import com.labsynch.labseer.chemclasses.CmpdRegSDFReader;
 import com.labsynch.labseer.exceptions.CmpdRegMolFormatException;
 
-import org.RDKit.SDMolSupplier;
-
 public class CmpdRegSDFReaderRDKitImpl implements CmpdRegSDFReader {
 
-    private SDMolSupplier molSupplier;
-
-    public CmpdRegSDFReaderRDKitImpl(String fileName) {
-        this.molSupplier = new SDMolSupplier(fileName);
+    private Scanner scanner;
+
+    private ExternalStructureService bbChemStructureService;
+
+    public CmpdRegSDFReaderRDKitImpl(String fileName, ExternalStructureService bbChemStructureService) {
```

Comment:
Was there a technical limitation why this couldn't be Autowired and needs to be passed in as an argument?

Reply:
Gap in my knowledge or possibly a technical limitation.  This was the only way I could get passed down in the right way to new up a `CmpdRegMoleculeRDKitImpl` which would still use bbchem service.